### ISSUE TITLE
Support for later pytorch and cuda versions

### ISSUE THREAD

**Issue by Stfort52** [BayraktarLab/cell2fate#21]
First off, thanks a lot for the great work on this tool!

As far as I know, the current setup is using PyTorch 1.11.0, which limits compatibility with newer CUDA versions. I was wondering if there are any plans to upgrade PyTorch to a more recent release (e.g., 2.x), to take advantage of the latest features and broader support for newer CUDA toolkits?

Thanks again!

**Comment by AlexanderAivazidis** [comment]
Hi, thanks for the issue. I think to make cell2fate compatible with recent CUDA versions you can install a newer pytorch like this, after having installed cell2fate:

conda activate cell2fate_env
pip install torch torchvision --extra-index-url https://download.pytorch.org/whl/cu116

This works for me on a GPU with CUDA 12.6 . Everything, including model training seems to run fine, even with a different pytorch version. I have added a note about this to the installation instructions.

**Comment by Stfort52** [comment]
Thanks for your comment!

I was worried about bumping torch version up, as it was fixed as 1.11.0 in the package.
I'll try it on CUDA 12.1.

**Comment by Stfort52** [comment]
I tried it on CUDA 12.4. Although the numbers and the order of gene modules were a bit different, I still got similar velocity figures.

If you're interested, check it out.
https://gist.github.com/Stfort52/118070c4c21f35ca1c76298088955d33

Thanks again!
